src/shared.js

```diff
@@ -834,21 +834,31 @@ export const removeBag = async(boxId, bags) => {
 }
 
 /**
- * Fetches biospecimen collection data from the database, and removes '0008', '0009' and deviation tubes from each collection
- * @returns {Array} List of biospecimen collections
+ * Fetches biospecimen collection data from the database
+ * @returns {Promise}
  */
-export const searchSpecimenInstitute = async () => {
-    //https://us-central1-nih-nci-dceg-connect-dev.cloudfunctions.net/biospecimen?api=searchSpecimen
+export const searchSpecimenInstitute = async() => { // rename to searchSpecimenInstitute
     const idToken = await getIdToken();
     const response = await fetch(`${api}api=searchSpecimen`, {
         method: "GET",
         headers: {
             Authorization:"Bearer "+idToken
         }
     });
-    let a = await response.json();
+    return response.json();
+}
+
+/**
+ * Fetches biospecimen collection data from the database, and removes '0008', '0009' and deviation tubes from each collection
+ * @returns {Array} List of biospecimen collections
+ */
+export const filterSpecimenCollectionList = async () => {
+    //https://us-central1-nih-nci-dceg-connect-dev.cloudfunctions.net/biospecimen?api=searchSpecimen
+    const searchSpecimenInstituteResponse = await searchSpecimenInstitute()
+    const searchSpecimenInstituteList = searchSpecimenInstituteResponse?.data ? searchSpecimenInstituteResponse.data : []
```

Comment:
I'm assuming this is your method of error handling a 400 code / error resulting from searchSpecimenInstitute() ?

Reply:
This is correct. I'm not sure if this is extra or needed. There are places in our code where I followed the approach of parsing the JSON data assuming no bad status codes are returned. In your opinion is this error handling necessary?